test/t/thread/test_pool.cpp

```diff
@@ -70,3 +70,30 @@ TEST_CASE("thread") {
 
 }
 
+TEST_CASE("thread (user-provided pool)") {
+
+    auto pool_ptr = osmium::thread::Pool::create_instance(7);
+    auto& pool = *pool_ptr;
+
+    SECTION("can get access to thread pool") {
+        REQUIRE(pool.queue_empty());
+    }
+    
+    SECTION("can access user-provided number of threads") {
+        REQUIRE(pool.num_threads() == 7);
+    }
+
+    SECTION("can send job to thread pool") {
+        auto future = pool.submit(test_job_with_result {});
```

Comment:
The formatting is not mine, [it was there](https://github.com/osmcode/libosmium/blob/6a17a8ddd58c6d574259ac839fb550540e017eee/test/t/thread/test_pool.cpp#L60).